docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,461 @@
+course if each node is started manually then it's unlikely that too many nodes
+are started, but it's certainly possible to get into this situation if using a
+more automated orchestrator, particularly if a network partition happens at the
+wrong time.
+
+We avoid this by requiring a separate _cluster bootstrap_ process to take place
+on every brand-new cluster. This is only required the first time the whole
+cluster starts up: new nodes joining an established cluster can safely obtain
+all the information they need from the elected master, and nodes that have
+previously been part of a cluster will have stored to disk all the information
+required when restarting.
+
+The simplest way to bootstrap a cluster is to use the
+`elasticsearch-bootstrap-cluster` command-line tool:
+
+[source,txt]
+--------------------------------------------------
+$ bin/elasticsearch-bootstrap-cluster --failure-tolerance 1 \
+    --node http://10.0.12.1:9200/ --node http://10.0.13.1:9200/ \
+    --node https://10.0.14.1:9200/
+--------------------------------------------------
+
+The arguments to this tool are the target failure tolerance of the cluster and
+the addresses of (some of) its master-eligible nodes.
+
+If it is not possible to use this tool, you can also bootstrap the cluster via
+the API as described here. There are two steps to the bootstrapping process.
+Firstly, after all the nodes have started up, created their persistent node
+IDs, and discovered each other, the first step is to request a bootstrap
+document:
+
+[source,js]
+--------------------------------------------------
+# Return the current bootstrap document immediately
+GET /_cluster/bootstrap
+# Wait until the node has discovered at least 3 nodes, or 60 seconds has elapsed,
+# and then return the bootstrap document
+GET /_cluster/bootstrap?wait_for_nodes=3&timeout=60s
+--------------------------------------------------
+// CONSOLE
+
+The boostrap document contains information that the cluster needs to start up,
+and looks like the following.
+
+[source,js]
+--------------------------------------------------
+{
+  "master_nodes_failure_tolerance": 1,
+  "master_nodes":[
+    {"id":"USpTGYaBSIKbgSUJR2Z9lg"},
+    {"id":"gSUJR2Z9lgUSpTGYaBSIKb"},
+    {"id":"2Z9lgUSpTgSUYaBSIKbJRG"}
+  ]
+}
+--------------------------------------------------
+
+It is safe to repeatedly call `GET /_cluster/bootstrap`, and to call it on
+different nodes concurrently. This API will yield an error if the receiving
+node has already been bootstrapped or has joined an existing cluster.
+
+Once a bootstrap document has been received, it must then be sent back to the
+cluster to finish the bootstrapping process as follows:
+
+[source,js]
+--------------------------------------------------
+# send the bootstrap document back to the cluster
+POST /_cluster/bootstrap
+{
+  "master_nodes_failure_tolerance": 1,
+  "master_nodes":[
+    {"id":"USpTGYaBSIKbgSUJR2Z9lg"},
```

Comment:
maybe we could also (in addition) return the node names here so that it's clearer to a user which nodes we are talking about.

Reply:
Nice idea. I added this.